docs/import/add_disaster.js

```diff
@@ -0,0 +1,253 @@
+ * @param {Array<string>} states array of state (abbreviations)
+ * @return {Promise<boolean>} returns true after successful write to firestore.
+ */
+function writeNewDisaster(disasterId, states) {
+  if (disasters.has(disasterId)) {
+    setStatus('Error: disaster with that name and year already exists.');
+    return Promise.resolve(false);
+  }
+
+  disasters.set(disasterId, states);
+  toggleState(disasterId);
+
+  clearStatus();
+  const disasterOptions = $('#disaster > option');
+  // comment needed to quiet eslint on no-invalid-this rules
+  disasterOptions.each(/* @this HTMLElement */ function() {
+    if ($(this).val() > disasterId) {
+      $(createOptionFrom(disasterId)).insertBefore($(this));
+      return false;
+    }
+  });
+  $('#disaster').val(disasterId);
+  return getFirestoreRoot()
+      .collection('disaster-metadata')
+      .doc(disasterId)
+      .set({states: states})
+      .then(() => true);
+}
+
+/**
+ * Onclick function for the new disaster form. Writes new disaster to firestore,
+ * local disasters map and disaster picker. Doesn't allow name, year or states
+ * to be empty fields.
+ * @return {Promise<boolean>} resolves true if new disaster was successfully
+ *     written.
+ */
+function addDisaster() {
+  const year = $('#year').val();
+  const name = $('#name').val();
+  const states = $('#states').val();
+
+  if (!year || !name || !states) {
+    setStatus('Error: Disaster name, year, and states are required.');
+    return Promise.resolve(false);
+  }
+  if (isNaN(year)) {
+    setStatus('Error: Year must be a number.');
+    return Promise.resolve(false);
+  }
+  const disasterId = year + '-' + name;
+  return writeNewDisaster(disasterId, states);
+}
+
+// Needed for testing :/
+const emptyCallback = () => {};
+
+// TODO: add functionality for adding assets to disaster records from these
+// pickers.
+/**
+ * Requests all assets in ee directories corresponding to given states.
+ * @param {Array<string>} states e.g. ['WA']
+ * @return {Promise<void>} completes when we've finished filling all state
+ * pickers.
+ */
+function getAssetsFromEe(states) {
+  return ee.data.listAssets(eeLegacyPathPrefix + 'states', {}, emptyCallback)
+      .then((result) => {
+        const folders = new Set();
+        for (const folder of result.assets) {
+          folders.add(folder.id.substring((gdEePathPrefix + 'states/').length));
+        }
+        const promises = [];
+        for (const state of states) {
+          const dir = eeLegacyPathPrefix + 'states/' + state;
+          if (!folders.has(state)) {
+            ee.data.createFolder(dir, false, emptyCallback);
+            promises.push(Promise.resolve([state, []]));
+          } else {
+            promises.push(
+                ee.data.listAssets(dir, {}, emptyCallback).then((result) => {
+                  const assets = [];
+                  if (result.assets) {
+                    for (const asset of result.assets) {
```

Comment:
do folders have ids? Do we need to filter them out, in case someone creates a subfolder? Should we display a warning to the user that subfolders won't be recursed into?

Reply:
Added a silent check that it's an image, imagecollection, or table. I'm not too too fussed about warning the user it won't load subdirectories because I think that's sort of clear from the basic nature of the asset picker that that wouldn't work. I also worry about a warning being spammy.

\+ added a test